### PR TITLE
Cache server config settings.

### PR DESCRIPTION
**Pull request by lhofhansl**
From a discussion on another PR... Cache the server setting in to avoid retrieving them over and over again.

### CODE REVIEW

File: src/clientiface.cpp [hunk 1]
```diff
@@ -101,8 +112,7 @@ void RemoteClient::GetNextBlocks (
 		return;
 
 	// Won't send anything if already sending
-	if(m_blocks_sending.size() >= g_settings->getU16
-			("max_simultaneous_block_sends_per_client"))
+	if(m_blocks_sending.size() >= m_max_simul_sends)
```
Comment (SmallJoker) [inline src/clientiface.cpp]: Space missing after `if` and misplaced bracket. Belongs to this line end.

File: src/clientiface.cpp [hunk 2]
```diff
@@ -158,18 +168,15 @@ void RemoteClient::GetNextBlocks (
...
 
 	/*
 		Check the time from last addNode/removeNode.
 
 		Decrease send rate if player is building stuff.
 	*/
 	m_time_from_building += dtime;
-	if(m_time_from_building < g_settings->getFloat(
-				"full_block_send_enable_min_time_from_building"))
+	if(m_time_from_building < m_min_time_from_building)
```
Comment (SmallJoker) [inline src/clientiface.cpp]: Space and bracket

**Comment by rubenwardy** [comment]
It would also be good to look for settings in this class cached by static, and correct them to be member cached

**Comment by lhofhansl** [comment]
Got tired of the code nits. Just went through clientiface.cpp and fixed all of them.
Generally not a friend of mixing things together.

No static configs used in this class.

File: src/clientiface.cpp [hunk 3]
```diff
@@ -113,7 +122,7 @@ void RemoteClient::GetNextBlocks (
 	LuaEntitySAO *lsao = getAttachedObject(sao, env);
 	const v3f &playerspeed = lsao? lsao->getVelocity() : player->getSpeed();
 	v3f playerspeeddir(0,0,0);
-	if(playerspeed.getLength() > 1.0*BS)
+	if (playerspeed.getLength() > 1.0*BS)
```
Comment (paramat) [inline src/clientiface.cpp]: 1.0f

File: src/clientiface.cpp [hunk 4]
```diff
@@ -145,8 +153,7 @@ void RemoteClient::GetNextBlocks (
 			<<m_nearest_unsent_reset_timer<<std::endl;*/
 
 	// Reset periodically to workaround for some bugs or stuff
-	if(m_nearest_unsent_reset_timer > 20.0)
-	{
+	if (m_nearest_unsent_reset_timer > 20.0) {
```
Comment (paramat) [inline src/clientiface.cpp]: 20.0f

File: src/clientiface.cpp [hunk 5]
```diff
@@ -406,7 +403,7 @@ void RemoteClient::SentBlock(v3s16 p)
 	if (m_blocks_modified.find(p) != m_blocks_modified.end())
 		m_blocks_modified.erase(p);
 
-	if(m_blocks_sending.find(p) == m_blocks_sending.end())
+	if (m_blocks_sending.find(p) == m_blocks_sending.end())
 		m_blocks_sending[p] = 0.0;
```
Comment (paramat) [inline src/clientiface.cpp]: 0.0f

**Comment by paramat** [comment]
Added line comments, also please could you check these files for other floats that should be specified with an 'f'?
Otherwise +1.

**Comment by lhofhansl** [comment]
Fixed even more style nits.

**Comment by paramat** [comment]
Thanks.

**Comment by paramat** [comment]
@nerzhul i see on IRC you were unhappy about something here http://irc.minetest.net/minetest-dev/2017-10-14#i_5111597 but the author cannot think of a better way, is there one?

**Comment by paramat** [comment]
:+1:

**Comment by nerzhul** [comment]
@paramat no choice currently, the settings should not be cached but instead used a nice std::array with enum fields to be fastest and reloaded when needed.
Then go

**Pull request merged**